src/TodoApp/Request.hs

```diff
@@ -52,95 +56,117 @@ data Request r where
   -- | Remove all tasks
   Reset :: Request ()
 
+
+data ConnectionType
+  = TCP
+  | UnixSocket FilePath
+
+data Connection = Connection {
+  uri :: URI,
```

Comment:
Why do you need a http URI when using unix sockets?

Reply:
When Unix socket is used, the HTTP URI is a dummy URI because `Network.HTTP.Req.req` requires a URI. 

I just realised that the type might be misleading here, so I changed it in https://github.com/juspay/todo-app/pull/20/commits/484939ab898ee1103fff9c39fd6515a700b0b290. Let me know if this makes sense

Reply:
Yes, this is better:

```haskell
data Connection
  = TCP URI
  | UnixSocket FilePath
```